node/blockRequestsQueue.go

```diff
@@ -21,38 +21,36 @@ import (
 	"github.com/alvalor/alvalor-go/types"
 )
 
-type transactionRequestsQueue interface {
+type blockRequestsQueue interface {
 	getData() map[string][]types.Hash
 }
```

Comment:
Some of the best developers in the world say that programming is basically about modelling data and transformations of data. In this context, I do not understand why you chose this way to model it. It seems overly complicated to me.

Reply:
Channel - used to model outgoing requests for the block headers. Queue to transform these outgoing requests into specific addr -> blocks map so that load is distributed evenly.

Reply:
We don't need to model outgoing requests in any way. What we need to queue are the actual network requests, as that's when the I/O happens. Anything up to that point can be figured out without delays.

Reply:
Agree, I double checked it and it looks like everything could be decided without the usage of the channel.